extension/intents/navigation/clickLink.js

```diff
@@ -0,0 +1,70 @@
+/* globals communicate, Fuse, log */
+
+this.followLink = (function() {
+  communicate.register("followLink", message => {
+    const { query } = message;
+    // Fuse options
+    const options = {
+      shouldSort: true,
+      tokenize: true,
+      matchAllTokens: true,
+      findAllMatches: true,
+      includeScore: true,
+      threshold: 0.1,
+      location: 0,
+      distance: 100,
+      maxPatternLength: 32,
+      minMatchCharLength: 3,
+      keys: [
+        {
+          name: "title",
+          weight: 0.8,
+        },
+        {
+          name: "text",
+          weight: 0.8,
+        },
+        {
+          name: "url",
+          weight: 0.2,
+        },
+      ],
+    };
+    const combinedContent = [];
+    for (const link of findLinks()) {
+      combinedContent.push({
+        element: link,
+        text: link.innerText,
+        title: link.getAttribute("title"),
+        url: link.url,
+      });
+    }
+    const fuse = new Fuse(combinedContent, options);
+    const matches = fuse.search(query);
+    if (!matches.length) {
+      return false;
+    }
+    let found;
+    for (const match of matches) {
+      const el = match.item.element;
+      return (found = el);
```

Comment:
The return here is definitely not correct. Try to keep this function and the loops simpler